framework/src/multiapps/TransientMultiApp.C

```diff
@@ -382,7 +382,9 @@ TransientMultiApp::solveStep(Real dt, Real target_time, bool auto_advance)
 
         // If we were looking for a steady state, but didn't reach one, we still need to output one
         // more time, regardless of interval
-        if (!at_steady)
+        // Note: if we turn off the output for all time steps for sub-scycling, we still need to
+        // have one output at the end.
+        if ((!at_steady && _detect_steady_state) || !_output_sub_cycles)
```

Comment:
Before outputting sub-cycling without steady-state detection will result into double outputting. I hate this complicated logic... `print_sub_cycles` makes things more complicated. If we do not have steady-state detection, we will see double screen (console) outputs by default. This change here does not fix the console output though which is the original purpose of this PR.

Reply:
I will give up this PR because it does not change the final outputted file. This should be fixed when we are solving #12742.

Reply:
@vincentlaboure when you create the moose test mimicking your real case, please also report minor output issues so that we can have a record to remind us to fix them. This double outputting on screen is one of them. Using `Moose::out` instead of `_console` in `MultiAppMeshFunctionTransfer.C` is another one, which results into some screen print-out of the level 1 multiapp not prefixed. Best.

Reply:
OK so now I'm a bit confused you don't want this merged? The _console fix was good but I'm also a little unclear as to the new logic.

Reply:
The logic is quite complicated. There are four booleans here: `_output_sub_cycles`, `_print_sub_cycles`, `at_steady` and `_detect_steady_state`.

1. `_output_sub_cycles` is set through a user parameter for outputs on sub-cycles, default to false;
2. `_print_sub_cycles` is set through a user parameter for screen-only outputs on sub-cycles, default to true;
3. `at_steady` is set by the code to true only when `_detect_steady_state` is true and steady-state is reached;
4. `detect_steady_state` is set through a user parameter for enabling stead-state detection.

I think this logic is correct `((!at_steady && _detect_steady_state) || !_output_sub_cycles)`, i.e. when not at steady-state or no output during sub-cycling, we force an output. The only thing bad is that if force-output is desired and users set `_print_sub_cycles` to true, we will see two duplicated screen outputs at the end. The old logic `(!at_steady)` is not very right because it does not take into account that when `_detect_steady_state` is not set to true by users, force-output will always happen.

I feel the logic is very complicated, so rather it to be fixed along with fixing #12742. Does this make sense?